### DISCUSSION TITLE
Design improvement discussion - Nav/Persistent UI Areas

### DISCUSSION THREAD

**Discussion by dzuk-mutant** [datafruits/datafruits#1342] [Ideas]
**This refers to the persistent UI areas on the website - both the navigation bar at the top, and the music/messages bar at the bottom.**

For this one I have a few vague ideas and observations of my own but I don't want to make presumptions about what is and is not important to people, so in this one I'm going to like, mention some thoughts of mine and ask folks what they think and develop ideas from there.

# Top nav
## There's a lot of stuff, maybe restructuring it would be a good idea
![image](https://github.com/datafruits/datafruits/assets/32108484/9af83388-71ca-4220-89ba-30fac0d7c168)

There are 14 page categories, 6 of them are in Radio, which leaves 8 nav sections on the top, which is kind of a lot IMO.

I think this area could be restructured to put more areas in more dropdowns so avoid presenting too much information at once.

## Some things that could be redesigned or removed
'$' is the support category, but it's small, almost hidden between '3D' and 'SHRIMPO'.

'3D' is our Mozilla Hub, which is no longer in service,

## Some things are more or less obvious than others, perhaps less than might be a good idea?
As a formerly-new user I found 'da fruit standz' and 'Fruitopiyeah' to be a bit too confusing and alienating.

A thought I had is that maybe using accompanying icons, they could be made to be more immediately understandable while keeping the in-jokes, or maybe they could be rewritten to be a bit more obvious as to what function they have without making them completely flat and dry descriptions.

## We can make the interface look less cramped by making the bar transparent in some ways
This is demonstrated in some prototypes I already presented, where I made the nav at least transparent when it's at the top, with the intention of perhaps matching it to the background when scrolled. It doesn't change the amount of space it takes, but it has the effect of making the interface feel more open.

![image](https://github.com/datafruits/datafruits/assets/32108484/4a4a7102-3871-433c-bce2-858d1e4183df)
![image](https://github.com/datafruits/datafruits/assets/32108484/5ec78aab-c568-4bec-8ce0-f901933f8f27)

## Radio dropdown arrow is finnicky to operate on desktop
It's a small area, it's only the arrow rather than the entire heading, and it doesn't disappear when your cursor moves away from the dropdown menu, taking up even more UI real estate.

# Bottom nav
## It's larger than it needs to be.
On full-screen 1080p, it looks like this:
![image](https://github.com/datafruits/datafruits/assets/32108484/c6c625a4-9eca-4184-90f5-c765be8c0ed9)

There's a lot of under-utilised space. at smaller sizes too.

One idea I had is to stop the music area from stretching the screen, and if the datafruits messages component still works best on the bottom of the screen (I've had some thoughts of maybe moving that around? IDK), that could sit next to the music component side by side instead. as a whole, both components in that layout would have a smaller footprint on the visuals of the website overall.

![image](https://github.com/datafruits/datafruits/assets/32108484/0bf49d7e-4824-4442-aab3-71d6bfd7346c)
Here's a really quick prototype of what a live stream view could look like if it wasn't forced to stretch the width of the screen.

## Buttons that aren't in the most appropriate places
There's 'Support' (a duplicate of the $ button on main nav) and 'hack' (a link to the GitHub repo). I'd argue that neither of these are relevant to the music player area and so should be moved to more relevant areas of the interface.

I think we could remove these two from that area, put a clearer Support button on the top nav and a link to the GitHub repo link somewhere in a new category on the main nav.

# General stuff
## Text is larger than I'd argue it needs to be by default.
For accessibility, we can make sure everything is sized in rems and folks can increase text size to increase UI size, but as a default, it takes up a lot of precious real-estate given all of the functionality we have. and it makes the middle content area smaller than it needs to be and cramps things up.

# One proposal for nav sections:
This is just me spitballing ^^' (I'm not proposing any section name changes here, just adding some extra detail as to what does what for the sake of discussion)

- Chat
- Radio
   - Timetable
   - Shows
   - Archives
   - DJ on Datafruits
- Community
    - Fruits
    - SHRIMPO
    - GitHub/Hack
    - da fruit standz (Forum)
    - Fruitopiyeah (Wiki)
- Stuff
   - Label
   - Shop
- About
- Support

**Comment by dudethatbe** [comment]
# top nav
> There are 14 page categories, 6 of them are in Radio, which leaves 8 nav sections on the top, which is kind of a lot IMO.

Completely agree that there's too much going on here, especially with some defunct services, and confusing names.

> We can make the interface look less cramped by making the bar transparent in some ways

🤯 I really do like the idea of a transparent navbar like that.  That could potentially solve the issue of video getting cut off (like you also mentioned in other chats).

 # bottom nav
 I agree its often taking up more space than it should and that a smaller player would be preferable. Is the yellow square meant to be a stop button or a placeholder for an image?

# bad buttons
Agreed about consolidating the support and $ button. We added the support button during a week-long show because people were asking how could they donate. And the $ button was added because the support button disappears on mobile. I'd associate "Support" and "Hack" to be in a sub-menu with "About". 

# big text
Refactoring text styles to be in rems seems doable 👍

# nav sections
We may need a dedicated issue just for this, but maybe "Chat" can be placed in "Community"? Then again I wouldn't want it to make it more difficult to find the chat 😅 I also think using "Forum" and "Wiki" are fine if we make these big changes. Some of these long names are here to take up all that under-utilized space in the first place.

**Reply by dzuk-mutant** [reply]
## Music player
In that concept, the yellow square would be show artwork. I thought it might be a cool idea since upcoming/live shows usually have their own artwork prepared beforehand.

## Chat
I personally feel like Chat is good not in a section because it's not just a chat really, but a listening/performance space for the entire radio station, and it impacts how the website behaves (like if there's viz on).

A part of me wishes there could be a better name for 'Chat' to emphasise this fact but I can't immediately think of anything that conveys it's full function that's succinct and accessible to newcomers.

Oooo. I just had an idea as I'm writing this - maybe the chat link could move into the music player (very prominently), then it would be much clearer where Chat belongs in the general way of things.

**Reply by dzuk-mutant** [reply]
![image](https://github.com/datafruits/datafruits/assets/32108484/0327cb63-4aec-4225-9926-9bc9965a2ac7)

...something like this! Imagine the datafruits messages being to the right of the music player 😅 

I feel like this move really helps semantically tie the chat with the live music more than having it in the top bar, and takes another segment off the top bar.

And the chat button could change state or collapse when the user is actually in the chat, maybe the music player controls could move altogether to make more room for viz.

**Reply by dudethatbe** [reply]
(meant to reply, not post a new thing) How To Display The Chat is another good candidate for a dedicated discussion. In the past, its been suggested to make the homepage actually the chat. Heck I even link to the /chat route in social media posts. Two persistent challenges for newcomers are (and this is just speaking from legacy experience):
* Finding the play button
* Joining the chat

To get back to the intent of this discussion, making the player smaller and using negative space to create a contrast _should_ solve that first issue. Everyone else (twitch, youtube) is smart and has a video play automatically when you load the page. We used to have auto-play, but y'know we only had audio and that was kinda bad practice at the time 🙃
Attaching the chat to the player is a good step too. I almost want to call this component the "datafruits-media-center" now that we're talking about embedding images.

**Comment by dzuk-mutant** [comment]
Here's an extremely radical concept - pushing that unified chat/media player concept to the top, with datafruits messages and put the site menu in a hamburger menu.

I don't have much of a working idea for how it will scale down! (or how I could wrap a stretchable SVG around a container lol) I just felt like trying it, especially given the mention of NTS in the aesthetic thread.

![image](https://github.com/datafruits/datafruits/assets/32108484/3a2464d7-20b8-4288-aaa9-a4f7de62e25b)

I like that this version makes it feel more chaotic, and it puts the live radio front and center.

**Reply by dzuk-mutant** [reply]
![image](https://github.com/datafruits/datafruits/assets/32108484/cea5cdda-41fd-418d-aa1a-fd994a1a2528)
And here's what it could possibly look like in chat when viz is active (not a real attempt at designing the chat)

**Reply by dzuk-mutant** [reply]
As mentioned on Discord, I feel like this nav concept might be the best balance so far of chaotic and accessible. I feel like maybe earlier concepts felt too clean and also didn't actually clean up the interface that much, while this one moves more things out of immediate view (making the entire bottom clear for content), while looking more silly and fun.

**Reply by dzuk-mutant** [reply]
![image](https://github.com/datafruits/datafruits/assets/32108484/bbff948a-4554-41bf-8074-cb2056a72e73)

What the hamburger menu could look like - it could have all the website sections uncollapsed, so it wouldn't create more interactions than having each website section as a dropdown would in the older concepts.

This also has the added benefit of giving the user a complete implicit picture of what Datafruits is by showing them everything that's on the website at once.

**Reply by dzuk-mutant** [reply]
note for later - I was thinking of making the logo/hamburger menu section a bit curvy and spiky like the leaves of a pineapple, but I'll leave that for another time if this concept is liked enough!

**Reply by dudethatbe** [reply]
That's a really great way of combining all of the main features into one navbar. Especially with putting the entire site map in a menu like that.

**Reply by dzuk-mutant** [reply]
More ideas on this front - like with the site navigation, the idea is to reduce the amount of discrete objects to interact with on-screen and group somewhat related things together. Here's what I had in mind for the user and settings area:

![image](https://github.com/datafruits/datafruits/assets/32108484/ff745df8-5cde-4bca-8063-5472619aa3c1)
![image](https://github.com/datafruits/datafruits/assets/32108484/6b836123-316c-4c50-9dd9-3f13823e86ce)

**Reply by dzuk-mutant** [reply]
![image](https://github.com/datafruits/datafruits/assets/32108484/0336d6af-a367-47be-bf43-494f1ed39d6e)

Some potential desktop collapsing

**Reply by dzuk-mutant** [reply]
![Screenshot 2024-06-13 113430](https://github.com/datafruits/datafruits/assets/32108484/30f1c563-cf2f-4f86-9b02-96c3e4bd770f)
![Screenshot 2024-06-13 113439](https://github.com/datafruits/datafruits/assets/32108484/80f05060-eea7-4f08-92d0-053956fad4ec)

And the reverse of collapsing. I was thinking to simply not expand the nav to fit the width of the screen once there's no benefit to expanding the contents and keep them as a little island of widgets in the top center of the screen.

![Screenshot 2024-06-13 113941](https://github.com/datafruits/datafruits/assets/32108484/980c74fa-8cf3-4a5c-81a7-037e3b9fda0e)

I was also thinking that they can just float above the contents of the web page when scrolling too.

**Comment by dzuk-mutant** [comment]
# A more detailed look into the widget nav concept!

![Screenshot 2024-06-14 214632](https://github.com/datafruits/datafruits/assets/32108484/59875760-f038-4413-a41d-ae092cda71a8)

![Screenshot 2024-06-14 214930](https://github.com/datafruits/datafruits/assets/32108484/731d8d6b-9118-4a9a-95f8-b0101846cab1)

![Screenshot 2024-06-14 214702](https://github.com/datafruits/datafruits/assets/32108484/a5da4139-a7ef-488f-94a6-6f75dd233496)

![Screenshot 2024-06-14 214647](https://github.com/datafruits/datafruits/assets/32108484/5e8fa575-4837-43e7-8b55-3d82611b9d0e)

![Screenshot 2024-06-14 215019](https://github.com/datafruits/datafruits/assets/32108484/b07fc85f-bd3b-4a8e-a3b5-3bf3387a00ad)

**Comment by dzuk-mutant** [comment]
Something I noticed tonight - live shows with hard-coded air dates create conflicting information when being seen from another time zone.
eg. I saw a show labelled as 12th July 2024 and it's 13th July in the UK so I just assumed that it wasn't live.

A 'live' indicator for all shows, prerecorded and streamed would do a lot, but not having any conflicting information to the user is still pretty important IMO.
